### PR TITLE
Adding headers in STS Calls for Confused Deputy

### PR DESCRIPTION
**Pull request by saieshwarm**
Adding headers in STS Calls for Confused Deputy to accept account id and the source ARN

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by kddejong — APPROVED** [review]

File: .pylintrc [hunk 1]
```diff
@@ -23,4 +24,5 @@ indent-string='    '
 max-line-length=160
```
Comment (ericzbeard) [inline .pylintrc:24]: Can we add an exception to just the one line if it's not possible to break it down? 160 is very long.

**Review by ericzbeard — COMMENTED** [review]
Looks good, just one question about pylintrc

File: .pylintrc (hunk 1 above)
Comment (saieshwarm) [reply .pylintrc:24]: This exception is from a different PR(PR No:663).
https://github.com/aws-cloudformation/cloudformation-cli/pull/663/files#diff-ba293004ae8f18975051e825bd8d5c70137a8b5d42d16e0f7c9354aa634c446c

File: src/rpdk/core/boto_helpers.py [hunk 2]
```diff
@@ -32,17 +32,35 @@ def _known_error(msg):
...
+        and headers["account_id"]
+        and headers["source_arn"]
+    ):
+        # Inject headers through the event system.
+        def inject_confused_deputy_headers(params, **kwargs):
+            params["headers"]["x-amz-source-account"] = headers["account_id"]
+            params["headers"]["x-amz-source-arn"] = headers["source_arn"]
+
+        sts_client.meta.events.register("before-call", inject_confused_deputy_headers)
+        LOG.info(headers)
```
Comment (himanshs23) [inline src/rpdk/core/boto_helpers.py:56]: do we need to still log it

**Pull request merged**